### DISCUSSION TITLE
Question on PalmNormal.Roll

### DISCUSSION THREAD

**Discussion by hongkai0818** [leapmotion/UnityModules#1278] [Q&A] [answered]
Hi, previously in our code we have been using the following codes.
float pitch = hand.Direction.Pitch;
float yaw = hand.Direction.Yaw;
float roll = hand.PalmNormal.Roll;

However, in the latest version 6.0.0, these are no longer valid. May I know how we can obtain the roll angle of hand.PalmNormal? If we need to calculate manually, how can we obtain the coordinate of the leap motion? (X-Z plane)

**Comment by MattGrayUL** [comment, marked as answer]
Hi, thanks for the question and thanks for trying out 6.0.0

You can see how we calculated these by grabbing the code from Vector in the [Legacy Package](https://github.com/ultraleap/UnityPlugin/releases/tag/com.ultraleap.tracking.preview%2F6.0.0)

Here is a static utility class that will produce the same results from unity's Vector3:

```
using UnityEngine;

public static class Vector3Utils
{
    public static float GetRoll(this Vector3 v)
    {
        return (float)System.Math.Atan2(v.x, -v.y);
    }

    public static float GetPitch(this Vector3 v)
    {
        return (float)System.Math.Atan2(v.y, -v.z);
    }

    public static float GetYaw(this Vector3 v)
    {
        return (float)System.Math.Atan2(v.x, -v.z);
    }
}
```

**Reply by hongkai0818** [reply]
Thank you so much for your prompt response
